platform/darwin/src/MGLStyleValue.h

```diff
@@ -58,7 +78,16 @@ MGL_EXPORT
  @param stops A dictionary associating zoom levels with style values.
  @return An `MGLStyleFunction` object with the given interpolation base and stops.
  */
-+ (instancetype)valueWithInterpolationBase:(CGFloat)interpolationBase stops:(NSDictionary<NSNumber *, MGLStyleValue<T> *> *)stops;
++ (instancetype)valueWithInterpolationBase:(CGFloat)interpolationBase stops:(NSDictionary<NSNumber *, MGLStyleValue<T> *> *)stops __attribute__((deprecated("Use +cameraFunctionValueWithStopType:stops:options:")));
+
+// TODO: API docs
++ (instancetype)cameraFunctionValueWithStopType:(MGLStyleFunctionStopType)stopType stops:(NS_DICTIONARY_OF(id, MGLStyleValue<T> *) *)stops options:(nullable NS_DICTIONARY_OF(MGLStyleFunctionOption, id) *)options NS_SWIFT_NAME(cameraFunctionValue(stopType:stops:options:));
```

Comment:
The overridden Swift name notwithstanding, it feels awkward to name an initializer such that it bridges into Swift as `MGLStyleValue.cameraFunctionValue(…`, using the class method syntax, instead of `MGLStyleValue.init(…`, using the initializer syntax. For one thing, it means repetition at the call site, which Swift strives to avoid.

Naming the factory methods `+valueWith…` would have the unfortunate effect of obscuring the type (camera, source, etc.), which is important. To disambiguate the various factory methods at the call site, how about naming the selectors along the lines of `+valueWithStopType:cameraStops:options:`?